app/src/main/java/com/example/marvelapp/service/CharacterServiceImpl.kt

```diff
@@ -0,0 +1,32 @@
+package com.example.marvelapp.service
+
+import com.example.marvelapp.entity.Character
+import com.example.marvelapp.mapper.MarvelMapper
+import com.example.marvelapp.service.api.MarvelApi
+import com.example.marvelapp.service.request.generator.MarvelRequestGenerator
+import com.example.marvelapp.util.Result
+import org.koin.core.component.KoinComponent
+
+class CharacterServiceImpl(
+    private val api: MarvelRequestGenerator, private val mapper: MarvelMapper) : CharacterService, KoinComponent {
+
+    override fun getCharacters(): Result<List<Character>> {
+        try {
+            val callResponse = api.createService(MarvelApi::class.java).getCharacters()
+            val response = callResponse.execute()
+            if (response.isSuccessful)
+                response.body()?.let {
+                    mapper.transform(it)
+                }?.let {
+                    return Result.Success(it)
+                }
+        } catch (e: Exception) {
+            return Result.Failure(Exception(FAILED_CHARACTERS_GET))
+        }
+        return Result.Failure(Exception(FAILED_CHARACTERS_GET))
+    }
+
+    companion object {
+        private const val FAILED_CHARACTERS_GET = "COULDN'T FIND ANY CHARACTERS"
```

Comment:
if you will show this message on the screen, this has to be in a String resource

Reply:
fixed

Reply:
fixed